core/src/test/java/nl/nn/adapterframework/util/FileNameComparatorTest.java

```diff
@@ -0,0 +1,94 @@
+package nl.nn.adapterframework.util;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * FileNameComparator Tester.
+ *
+ * @author <Sina Sen>
+ */
+public class FileNameComparatorTest {
+
+
+	@Test
+	public void testComparator() throws Exception {
+		List<File> list = new ArrayList<>();
+		list.add(new File("test.txt"));list.add(new File( "teSt.txt"));
+		list.add(new File("0123.py")); list.add(new File("test.xml")); list.add(new File("document.txt"));
+		Collections.sort(list, new FileNameComparator());
+		assertEquals("0123.py", list.get(0).getName());
+		assertEquals("document.txt", list.get(1).getName());
+		assertEquals("teSt.txt", list.get(2).getName());
+		assertEquals("test.txt", list.get(3).getName());
+		assertEquals("test.xml", list.get(4).getName());
+
+	}
+
+	/**
+	 * Method: getNextIndex(String s, int start, boolean numericPart)
+	 */
+	@Test
+	public void testGetNextIndex() throws Exception {
+		int i = FileNameComparator.getNextIndex("test.txt", 0, true);
+		int j = FileNameComparator.getNextIndex("test.txt", 0, false);
+		assertEquals(0, i);
+		assertEquals(8, j);
+	}
```

Comment:
This is an edge case. In a filesystem it's not possible to have 2 files with the same name. Use test.txt twice theoretically cannot happen. Also don't test for an exact integer, but rather something equal, higher or lower then `0`.